File: teamscale-client/src/test/java/com/teamscale/client/ProxySystemPropertiesTest.java
package com.teamscale.client;

import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertThrows;

class ProxySystemPropertiesTest {

	private static ProxySystemProperties properties = new ProxySystemProperties(ProxySystemProperties.Protocol.HTTP);

	@AfterAll
	static void teardown() {
		properties.removeProxyPort();
	}

	@Test
	void testPortParsing() {
		properties.setProxyPort(9876);
		assertThat(properties.getProxyPort()).isEqualTo(9876);
		properties.setProxyPort("");
		assertThat(properties.getProxyPort()).isEqualTo(-1);
		String incorrectFormatValue = "nonsense";
		properties.setProxyPort(incorrectFormatValue);
		ProxySystemProperties.IncorrectPortFormatException exception = assertThrows(ProxySystemProperties.IncorrectPortFormatException.class,
				properties::getProxyPort);
		assertThat(exception.getMessage()).isEqualTo(String.format("Could not parse proxy port \"%s\" set via \"%s\"", incorrectFormatValue, properties.getProxyPortSystemPropertyName()));
		properties.removeProxyPort();
		assertThat(properties.getProxyPort()).isEqualTo(-1);
	}

}

File: teamscale-client/src/test/kotlin/com/teamscale/client/ProxySystemPropertiesTest.kt
package com.teamscale.client

import org.assertj.core.api.Assertions
import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.AfterAll
import org.junit.jupiter.api.Test

internal class ProxySystemPropertiesTest {
	@Test
	fun testPortParsing() {
		properties.proxyPort = 9876
		assertThat(properties.proxyPort).isEqualTo(9876)
		assertThatThrownBy {
			properties.proxyPort = 0
		}.hasMessage("Port must be a positive integer")
		assertThatThrownBy {
			properties.proxyPort = 65536
		}.hasMessage("Port must be less than or equal to 65535")
		properties.clear()
		assertThat(properties.proxyPort).isEqualTo(-1)
	}

	companion object {
		private val properties = ProxySystemProperties(ProxySystemProperties.Protocol.HTTP)

		@JvmStatic
		@AfterAll
		fun teardown() {
			properties.clear()
		}
	}
}